search/src/Components/upload/upload_stepper.js

```diff
@@ -299,6 +288,39 @@ class UploadStepper extends React.Component {
         }
     }
 
+    isNextEnabled() {
+        const step = stepsByType[this.props.upload_type][this.state.activeStep].type
+        const currentStageValid = (step) => {
+            switch (step) {
+                case "categories":
+                    return this.state.categories_saved
+                case "agcontext":
+                    return this.state.coco_form_validation['agcontexts']
+                case "metadata":
+                    return this.state.coco_form_validation['metadata']
+                case "images":
+                    return this.state.imageReady
+                default:
+                    return true
+            }
+        }
```

Comment:
Can we just have a single `state` property called `stepValid`? We could just use `setState({stepValid: xyz}) in each handler.

Reply:
I don't think it's a better way to solve it, as it may mess between different steps and harder to maintain. If users have already got validated in the agcontexts and metadata stage and he wants to make some changes about categories, then events are needed to set stepValid back and forth.

Reply:
Fine, so let's have `this.state.stepValid = {}` and use:
```
    const currentStageValid = this.state.stepValid[step];
```

Reply:
See #464 for an example